core/vm/contracts.go

```diff
@@ -26,6 +27,7 @@ import (
 	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/crypto/bn256"
 	"github.com/ethereum/go-ethereum/params"
+	"github.com/keep-network/blake2b/compression"
```

Comment:
Please move that file into the `crypto` package (I already asked on the previous PR). i don't want to add a dependency to an external project just to expose a method from Go's crypto lib. It's fine to embed it among the other crypto libs we already have embedded into our code.

Reply:
To be clear, this is not vanilla Go crypto lib code. Go code does not expose `F` compression function, and we had to modify it to allow for an arbitrary number of rounds, have the interface compatible with what we expect from the precompile and removed some pieces that are not really a part of compression but rather hashing. Please compare:
https://github.com/keep-network/blake2b/blob/master/compression/f.go#L39
https://github.com/golang/crypto/blob/master/blake2b/blake2b_generic.go#L30

Embedding F compression code in `go-ethereum` was my initial approach but there is another complication here. The only official test vector for the `F` function is [the one from BLAKE2b RFC Appendix A](https://github.com/keep-network/blake2b/blob/master/compression/f_test.go#L37). All other test vectors are for the [entire hash algorithm](https://github.com/keep-network/blake2b/blob/master/hash_test.go#L283) (which includes `F` as one of its parts). If I move that file to the `crypto` package, we'll lose the coverage by those vectors and stay with the one from Appendix A. I can move the entire reworked BLAKE2b code here (including hash test vectors) but I am not sure if it's the right option since we'll end up with an unused code.

I am not sure what's the best option here. Of course, leaving the decision to you.

Reply:
@karalabe I will move that file along with [a test](https://github.com/keep-network/blake2b/blob/master/compression/f_test.go#L36) to `crypto` package. What do you think about leaving [`toEthereumTestCase`](https://github.com/keep-network/blake2b/blob/master/compression/f_test.go#L66) code there? It generates test input/output in the format expected by the precompile based on `F` test vector. I personally have mixed feelings about leaving it.

Reply:
Honestly, I think it's fine to leave the test case as is, and maybe hard code the ethereum version too if you want to duplicate it. This isn't code that we expect to change, neither are the test vectors.

Reply:
👍 Done!